pandas/tests/frame/test_timeseries.py

```diff
@@ -366,6 +366,19 @@ def test_operation_on_NaT(self):
         exp = pd.Series([pd.NaT], index=["foo"])
         tm.assert_series_equal(res, exp)
 
+    def test_datetime_assignment_with_NaT_and_diff_time_units(self):
+        # GH 7492
+        data_ns = np.array([1, 'nat'], dtype='datetime64[ns]')
+        result = pd.Series(data_ns).to_frame()
+        result['new'] = data_ns
+        expected = pd.DataFrame({0: data_ns, 'new': data_ns})
+        tm.assert_frame_equal(result, expected)
+        # OutOfBoundsDatetime error shouldn't occur
+        data_s = np.array([1, 'nat'], dtype='datetime64[s]')
+        result['new'] = data_s
+        expected = pd.DataFrame({0: data_ns, 'new': data_s})
```

Comment:
Probably not needed to assert the original bug, but I would create the expected frame independently of `data_s` (eg `pd.DataFrame({0:[1e9, None], 'new': [1e9, None]}, dtype='datetime64[ns]')`)